spec/abstractops.html

```diff
@@ -1041,6 +1041,8 @@ <h1>ISO 8601 grammar</h1>
           TimeZoneIANANameComponent `/` TimeZoneIANANameTail
 
       TimeZoneIANALegacyName :
+          &gt; An ASCII-case-insensitive match for one of the following:
```

Comment:
I don't think this is valid notation, cf. https://tc39.es/ecma262/multipage/notational-conventions.html#sec-notational-conventions . There are probably many alternative approaches, but the one that comes to mind for me is providing a broad general definition of |TimeZoneIANALegacyName| (e.g., `TimeZoneIANALegacyName ::: TimeZoneIANANameTail[+Legacy]` where the Legacy parameter is passed down ultimately to |TZChar| for ```[+Legacy] DecimalDigit``` and ``` [+Legacy] `+` ``` expansions), using early errors or a `[> but only …]` [prose assertion](https://rbuckton.github.io/grammarkdown/usage/syntax.html#assertions) to restrict it to case-insensitive matches of this finite list, and adding an ambiguity-avoiding "but not TimeZoneIANANameTail" restriction in the reference to it from |TimeZoneIANAName|.

Reply:
Ugh, I was hoping that the prose description given here would be a sufficiently unambiguous description of what this production can match. This suggestion makes the grammar less easy to understand in my opinion... but I've tacked it on in an extra commit, see what you think of it. Having seen both, I'd personally prefer to leave it as it was.

I used a prose assertion instead of an early error here to restrict it to the specific legacy names; I find the notation less readable, but on the other hand more readable that it's right there in |TimeZoneLegacyIANAName| rather than at the end where an early error would be. It doesn't seem to me that the "but not TimeZoneIANANameTail" is necessary, because if |TimeZoneLegacyIANAName| is restricted to a finite list, it can't be ambiguous with TimeZoneIANANameTail[~Legacy].

Reply:
@gibson042 Let me know what you think of this solution. I'd rather keep it the way it was, but if you think this is an improvement then I'm fine with that too.

Reply:
[According to ECMA-262,](https://tc39.es/ecma262/multipage/notational-conventions.html#prod-grammar-notation-SourceCharacter), "a few nonterminal symbols are described by a descriptive phrase in sans-serif type in cases where it would be impractical to list all the alternatives" — I think this sufficiently describes the original prose description I had; I find the current solution in this PR impractical. I'll propose that we go ahead and merge this with the original prose description unless I hear any objections soon, since Frank is waiting on it.